Software/spartan/src/sensors/sensor.h

```diff
@@ -29,6 +31,7 @@ namespace spartan {
         // Data getters
         int getBusID() const;
         int getInstance() const;
+		const long frequency = 1000; 
```

Comment:
Specifying this variable that represents the frequency/period of polling should likely not be something we want to be in the `public` access section of our base `Sensor` class. By doing so, we allow any other portion of the FSW outside of the class to freely modify this frequency for any of the instantiated `Sensor` child objects. This could be error-prone, as we expect only the `Sensor` class and (potentially) its derived classes to modify this `frequency` variable.

In other words, this is something that should only be handled by the internal methods of the `Sensor` base class, since we want to ensure that we're manipulating this frequency in an expected and safe manner. At this point, we expect this frequency to be defined by compile-time, and it should be used when instantiating a new object for a `Sensor`-derived class. If we want to support run-time frequency changes, we should only allow access of this private member variable through a `protected` "setter" method that can be exclusively accessed by `Sensor` and its derived classes.

Here's my quick thoughts:
- Move this `frequency` variable from `public` to `private`.
- Add some accessor functions (you might not even need these depending on how you implement/what we want), including a "getter" and a "setter". Let's just start with the "getter" as `public`, and the "setter" as a `protected` method (this will support potential dynamic frequencies in the future!).
- Rename this `frequency` variable to more accurately represent the data it represents (refer to previous comment I made on this review).